Added alt attribute to <img> tag and added note (#1213)
withastro/docs: src/pages/en/guides/imports.md

import txtReference from './words.txt'; // txt === '/src/words.txt'

// This example uses JSX, but you can use import references with any framework.
<img src={imgReference} alt="image description" />;
```

All other assets not explicitly mentioned above can be imported via ESM `import` and will return a URL reference to the final built asset. This can be useful for referencing non-JS assets by URL, like creating an image element with a `src` attribute pointing to that image.

It can also be useful to place images in the `public/` folder as explained on the [project-structure page](/en/core-concepts/project-structure/#public).

:::note

> Adding **ALT text** to `img` tags is awesome practice! Don't forget to add `alt` text to your images, or just leave the attribute empty if the image is purely decorative.

Reviewer: ```suggestion
Adding **alt text** to `<img>` tags is encouraged for accessibility! Don't forget to add an `alt="a helpful description"` attribute to your image elements. You can just leave the attribute empty if the image is purely decorative.
```
Thanks for this helpful (and, encouraged!) addition @Njong392! 🥳 

I've made some suggestions to consider, and this looks like a lot  😅 ... it's not that I'm super picky (ok... maybe I HAVE BEEN KNOWN to be picky, lol) it's just that I do like to explain the thoughts I have as I go through text so that people know why I might suggest changing even just a single word sometimes... that it's not just to pick at things.

So, here's what went through my mind with this. For your consideration; let me know what you think!

- it might seem redundant, but we do typically use the tag symbols in addition to the word when we reference elements. So I've added those for consistency.

- I checked a handful of websites, and most of the places I've seen use "alt text" in all lower case. Some use "Alt Text" but the lower case seems more common? If there's a canonical source you're aware of, please let me know! Otherwise, I think with the bold text already, we should probably just go with "alt text" lower case.

- It **is** AWESOME practice, but I think I like the idea of specifically stating why, and reinforcing that we shoudld care about accessibility, even beyond just alt text!

- I also similarly wanted to remove "or", because the suggestion should be include the tag no matter what, right? So, let's show them the attribute, and then separately say, "No, srsly, write the attribute! If it's purely decorative, we can just leave the alt text blank!"

Author: Ohh... you made the PR look a lot better :) I do understand your changes, and I think it reads a lot better with what you added. As per whether it's "alt text" or "Alt Text" or "ALT Text", I admit that I have never really paid much attention haha

Also, I just wanted to mention this. I did see someone mention before that alt texts could also be added to SVGs. Stuff like videos and GIFs too. Would it be too much to not only mention img tags in this situation? I get that the context of the note refers to the `<img>` tag mentioned in the docs, but is it worth mentioning the others?

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :question: I have a question

Reviewer: (It's my responsibility to pay ALL the attention to ALL the things.....) ;)

OK, with all the other elements, too.... in that case what about something like:

> Don't forget to add an `alt="a helpful description"` attribute to all of your media elements

(Basically replacing "image" with "media"?) Or, what other generic term could we use there? We could even tack a list on the end like:

> Don't forget to add an `alt="a helpful description"` attribute to all of your media elements, including SVGs, videos and GIF too.

What about doing something like that?

Reviewer 2: > Also, I just wanted to mention this. I did see someone mention before that alt texts could also be added to SVGs. Stuff like videos and GIFs too. Would it be too much to not only mention img tags in this situation? I get that the context of the note refers to the <img> tag mentioned in the docs, but is it worth mentioning the others?

Hi @Njong392 👋, that's a great question!

GIFs also use the `img` tag so I think we don't need to explicitly mention them. About videos and SVGs, they shouldn't be described with `alt` because their respective elements don't accept it, according to [W3Schools](https://www.w3schools.com/tags/att_img_alt.asp) and [MDN](https://developer.mozilla.org/pt-BR/docs/Web/HTML/Attributes).

To give more context, they have their own accessibility recommendations:
* Non-decorative SVGs should be described with `aria-label` or `aria-labelledby`.
* Videos need captions (if they contain audio) and/or be described in the following paragraph.

Author: Oh... I see where you are coming from. That is true! I now wonder how the note could be altered to fit every media type (if we should include every media type.)

@sarah11918 how about

> Don't forget to add a text-based description to all of your media elements (from images to SVGs, to GIFs, and videos)

Reviewer 2: > I now wonder how the note could be altered to fit every media type (if we should include every media type.)

I personally think we don't need to mention every media type here, just image elements is fine I think. The accessibility patterns and gotchas vary a lot depending on the media and how it's used, so properly explaining this would be a guide of its own. What I think is our big objective here is to make our docs' readers aware of accessibility and incentivize them to know more about it rather than being an in-depth resource about it.

Reviewer: Yes, I agree with @Yan-Thomas here. It's a BIG topic that is not really in our scope, and if Yan thinks it's fine to keep it to image elements (since, that's what we're talking about here!) then I'm 100 on board with that! 

So, I would say all my original suggestions are still my takes on this! @Njong392, if you like those, you can click to "commit" the suggested comment as is. Or, rewrite to suit your taste and I'll look over it again.

Author: > So, I would say all my original suggestions are still my takes on this! @Njong392, if you like those, you can click to "commit" the suggested comment as is. Or, rewrite to suit your taste and I'll look over it again.

Awesome! I initially already agreed to your original suggestions, so I'll just go ahead and commit those :)

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix
